module/VuFindTheme/src/VuFindTheme/View/Helper/ThemeResources.php

```diff
@@ -0,0 +1,186 @@
+            $this->addMetaTags();
+            $this->addLinks();
+            $this->addScripts('header');
+            return null;
+        }
+        if ($position == 'footer') {
+            return $this->addScripts('footer');
+        }
+    }
+
+    /**
+     * Add meta tags to header.
+     *
+     * @return void
+     */
+    protected function addMetaTags()
+    {
+        // Set up encoding:
+        $headMeta = $this->getView()->plugin('headMeta');
+        $headMeta()->prependHttpEquiv(
+            'Content-Type', 'text/html; charset=' . $this->container->getEncoding()
+        );
+
+        // Set up generator:
+        $generator = $this->container->getGenerator();
+        if (!empty($generator)) {
+            $headMeta()->appendName('Generator', $generator);
+        }
+    }
+
+    /**
+     * Add links to header.
+     *
+     * @return void
+     */
+    protected function addLinks()
+    {
+        // Convenient shortcut to view helper:
+        $headLink = $this->getView()->plugin('headLink');
+
+        // Load CSS (make sure we prepend them in the appropriate order; theme
+        // resources should load before extras added by individual templates):
+        foreach (array_reverse($this->container->getCss()) as $current) {
+            $parts = $this->container->parseSetting($current);
+            // Special case for media with paretheses
+            // ie. (min-width: 768px)
+            if (count($parts) > 1 && substr($parts[1], 0, 1) == '(') {
+                $parts[1] .= ':' . $parts[2];
+                array_splice($parts, 2, 1);
+            }
+            $headLink()->forcePrependStylesheet(
+                trim($parts[0]),
+                isset($parts[1]) ? trim($parts[1]) : 'all',
+                isset($parts[2]) ? trim($parts[2]) : false
+            );
+        }
+
+        // Compile and load LESS (make sure we prepend them in the appropriate order
+        // theme resources should load before extras added by individual templates):
+        foreach (array_reverse($this->container->getLessCss()) as $current) {
+            $parts = $this->container->parseSetting($current);
+            $headLink()->forcePrependStylesheet(
+                $headLink()->addLessStylesheet(trim($parts[0])),
+                isset($parts[1]) ? trim($parts[1]) : 'all',
+                isset($parts[2]) ? trim($parts[2]) : false
+            );
+        }
+
+        // If we have a favicon, load it now:
+        $favicon = $this->container->getFavicon();
+        if (!empty($favicon)) {
+            $imageLink = $this->getView()->plugin('imageLink');
+            $headLink(
+                [
+                    'href' => $imageLink($favicon),
+                    'type' => 'image/x-icon', 'rel' => 'shortcut icon'
+                ]
+            );
+        }
+    }
+
+    /**
+     * Add scripts to header or footer.
+     *
+     * @param string $position 'header' or 'footer'.
+     *
+     * @return string|null
```

Comment:
I find this dual behavior a bit confusing. Would it make sense to create a FooterScript helper to handle scripts in footer? Not doing so would mean that footer scripts couldn't use the asset pipeline (be minimized), right?

Reply:
I hadn't considered this, but I see now you are correct that footer scripts will not go through the asset pipeline since they're using inlineScript instead of headScript (which, of course, makes sense given the context in which they are being rendered). I don't think there's an easy solution to that short of adding asset pipeline support to the inlineScript helper. Would it make sense to address that concern in a separate pull request?

I guess the other question is whether to change HeadThemeResources to ThemeResources as was done here, or to add FooterThemeResources. I don't have a strong opinion -- that approach might be more clear, but the current approach is more flexible if we have additional positions for scripts in the future. One key question is whether there is likely to be a third position in the future, or if there are only two possible scenarios to consider.

Reply:
Since i don't know the asset pipeline that well i can't really take part in the discussion... Please let me know if i can implement any concrete optimizations. Maybe you would like to discuss this topic at the next community call on March 2nd, unfortunately i won't be able to attend this time.

Reply:
@mtrojan-ub, I wonder if the problem here is that this pull request is solving two problems at once (the ordering of JS files, as well as the header vs. footer rendering of those files). I think it might be easier to discuss, review and ultimately merge this if we break it into two parts. I think we're very close to getting the advanced configuration/ordering logic merged, but I think the issues @EreMaijala raised require some more thought and discussion on the header vs. footer issue.

How would you feel if I opened a new PR containing only the relevant changes to the ResourceContainer and theme configuration? If we can get that fully tested and merged, then it will reduce this PR to only the positioning parts, and those may be easier to talk about if there's less unrelated code coexisting with them. If you think it's worth the effort, I'm happy to give it a try.

Reply:
That sounds good! Let me know if i can help you in any way.

Reply:
Great -- see #1863. I'll formally request a review from you there as soon as I finish running the test suite to make sure I did this right... :-)